### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by arjankowski**

### CODE REVIEW

File: Sources/Core/ThreadSafeQueue.swift [hunk 1]
```diff
@@ -30,13 +30,13 @@ class ThreadSafeQueue<T> {
...
+            }
+
             self.completionQueue.async {
-                if !self.list.isEmpty {
-                    completion(self.list.removeFirst())
-                }
-                else {
-                    completion(nil)
-                }
+                completion(firstItem)
```
Comment (PJSimon) [inline Sources/Core/ThreadSafeQueue.swift:39]: Can you explain this change, please?

File: Sources/Modules/SearchModule.swift [hunk 2]
```diff
@@ -23,9 +23,9 @@ public enum SearchScope: BoxEnum {
     /// - Parameter value: String representing search scope
     public init(_ value: String) {
         switch value {
-        case "user":
+        case "user_content":
             self = .user
-        case "enterprise":
+        case "enterprise_content":
```
Comment (PJSimon) [inline Sources/Modules/SearchModule.swift:28]: Nice catch!  :^)

File: Sources/Responses/EventContainer.swift [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import Foundation
 
 /// Container for Events
+@available(*, deprecated, message: "This class is obsoleted and will be removed in v5.0. Please use EntryContainer<Element> instead.")
```
Comment (PJSimon) [inline Sources/Responses/EventContainer.swift]: Can you briefly explain this one, please?

File: Sources/Responses/EventType.swift [hunk 4]
```diff
@@ -452,6 +452,36 @@ public enum EventType: BoxEnum {
...
+        case "DEVICE_TRUST_CHECK_FAILED":
+            self = .deviceTrustCheckFailed
+        case "TERMS_OF_SERVICE_ACCEPT":
+            self = .termsOfServiceAccepted
+        case "TERMS_OF_SERVICE_REJECT":
+            self = .termsOfServiceRejected
+        case "DOWNLOAD":
+            self = .downloaded
+        case "DELETE_USER":
+            self = .deletedUser
```
Comment (PJSimon) [inline Sources/Responses/EventType.swift:484]: Is this related to fixing existing tests?  If not, maybe we should break this out into another PR for adding missing EventTypes?

File: Sources/Responses/Token.swift [hunk 5]
```diff
@@ -35,6 +35,6 @@ public class Token: BoxModel {
         accessToken = try BoxJSONDecoder.optionalDecode(json: json, forKey: "access_token")
         expiresIn = try BoxJSONDecoder.optionalDecode(json: json, forKey: "expires_in")
         tokenType = try BoxJSONDecoder.optionalDecode(json: json, forKey: "token_type")
-        restrictedTo = try BoxJSONDecoder.optionalDecodeCollection(json: json, forKey: "scope")
+        restrictedTo = try BoxJSONDecoder.optionalDecodeCollection(json: json, forKey: "restricted_to")
```
Comment (PJSimon) [inline Sources/Responses/Token.swift:38]: Great catch!

File: Sources/Responses/Webhook.swift [hunk 6]
```diff
@@ -202,9 +202,9 @@ public class Webhook: BoxModel {
             case .taskAssignmentUpdated:
                 return "TASK_ASSIGNMENT.UPDATED"
             case .metadataInstanceCreated:
-                return "METEDATA_INSTANCE.CREATED"
+                return "METADATA_INSTANCE.CREATED"
             case .metadataInstanceUpdated:
-                return "METEDATA_INSTANCE.UPDATED"
+                return "METADATA_INSTANCE.UPDATED"
```
Comment (PJSimon) [inline Sources/Responses/Webhook.swift:207]: Just double-checking: not a breaking change?

**Review by PJSimon — DISMISSED** [review]

File: Sources/Core/ThreadSafeQueue.swift (hunk 1 above)
Comment (arjankowski) [reply Sources/Core/ThreadSafeQueue.swift:39]: To ensure that ThreadSafeQueue is really safe, all operations like `enqueue`, `dequeue`, `isEmpty` and `peek` should be executed on the same serial queue which is in our case `utilityQueue`. 
Otherwise a race condition can occurs which happened in our case. 

In `ThreadSafeQueue` class `enqueue` and `isEmpty` methods are accessing our resource which is a `private var list = [T]()` property in `utilityQueue`, but `dequeue` and `peek`
are accessing to `list` property on different queue `completionQueue`. So it is possible that two separeted threads accessing our `list` property at the same time hence race condition.

To fix that I changed that all operations on `list` property are executed on `utilityQueue` and `completionQueue` only gets the result.

File: Sources/Modules/SearchModule.swift (hunk 2 above)
Comment (arjankowski) [reply Sources/Modules/SearchModule.swift:28]: thx!

File: Sources/Responses/EventContainer.swift (hunk 3 above)
Comment (arjankowski) [reply Sources/Responses/EventContainer.swift]: `EventContainer` and `TrackingCode` classes are not used in our code and should be removed from repository.
The @available attribute used here will inform the consumer of this class by displaying a warning message in his code with the message placed in `message` field.

File: Sources/Responses/EventType.swift (hunk 4 above)
Comment (arjankowski) [reply Sources/Responses/EventType.swift:484]: I think that it should be placed it this commit. 
These values that we see here acutally were added a long time ago, but thanks to tests I found out that they are actually not working.
This is because there were misssing a mapping json(string) values into enum values, so I've just added it here.

File: Sources/Responses/Token.swift (hunk 5 above)
Comment (arjankowski) [reply Sources/Responses/Token.swift:38]: thx!

File: Sources/Responses/Webhook.swift (hunk 6 above)
Comment (arjankowski) [reply Sources/Responses/Webhook.swift:207]: Unfortunately, but because of this typo, the backend was receiving invalid values that it can not recognize

It's not a breaking change because it doesn't affect the existing API in any way.

File: Sources/Core/ThreadSafeQueue.swift (hunk 1 above)
Comment (PJSimon) [reply Sources/Core/ThreadSafeQueue.swift:39]: Cool - I see that now, on line 28!  Thanks!

**Review by PJSimon — APPROVED** [review]
